routers/web/repo/pull.go

```diff
@@ -1071,6 +1071,25 @@ func viewPullFiles(ctx *context.Context, specifiedStartCommit, specifiedEndCommi
 	ctx.Data["CanBlockUser"] = func(blocker, blockee *user_model.User) bool {
 		return user_service.CanBlockUser(ctx, ctx.Doer, blocker, blockee)
 	}
+	if !willShowSpecifiedCommit && !willShowSpecifiedCommitRange {
+		ctx.Data["SourcePath"] = pull.HeadRepo.Link() + "/src/branch/" + url.PathEscape(pull.HeadBranch)
+
+		if !pull.HasMerged && ctx.Doer != nil {
+			if err := pull.LoadHeadRepo(ctx); err != nil {
+				ctx.ServerError("LoadHeadRepo", err)
+				return
+			}
+			perm, err := access_model.GetUserRepoPermission(ctx, pull.HeadRepo, ctx.Doer)
+			if err != nil {
+				ctx.ServerError("GetUserRepoPermission", err)
+				return
+			}
+			ctx.Data["CanEditFile"] = perm.CanWrite(unit.TypeCode) || issues_model.CanMaintainerWriteToBranch(ctx, perm, pull.HeadBranch, ctx.Doer)
+			ctx.Data["EditFileTooltip"] = ctx.Tr("repo.editor.edit_this_file")
```

Comment:
Where is this variable used?

Reply:
They will be used on `box.tmpl` which is a party of the pull request files tab.